src/BusinessLogic/Resources/js/RegisterController.js

```diff
@@ -0,0 +1,232 @@
+if (!window.Packlink) {
+    window.Packlink = {};
+}
+
+(function () {
+    /**
+     * Handles register page logic.
+     *
+     * @param {{getRegistrationData: string, submit: string}} configuration
+     * @constructor
+     */
+    function RegisterController(configuration) {
+
+        const templateService = Packlink.templateService,
+            ajaxService = Packlink.ajaxService,
+            state = Packlink.state,
+            utilityService = Packlink.utilityService,
+            translationService = Packlink.translationService,
+            validationService = Packlink.validationService,
+            templateId = 'pl-register-page';
+
+        let form,
+            country;
+
+        /**
+         * The main entry point for controller.
+         */
+        this.display = additionalConfig => {
+            templateService.setCurrentTemplate(templateId);
+            country = additionalConfig.hasOwnProperty('country') ? additionalConfig.country : 'ES';
+
+            ajaxService.get(configuration.getRegistrationData, populateInitialValues);
+
+            const registerPage = templateService.getMainPage();
+
+            form = templateService.getComponent('pl-register-form', registerPage);
+            form.addEventListener('submit', register);
+
+            templateService.getComponent('pl-go-to-login', registerPage).addEventListener('click', goToLogin);
+
+            templateService.getComponent('pl-register-platform-country', registerPage).value = country;
+
+            initInputField('pl-register-email', validationService.validateEmail);
+            initInputField('pl-register-password', validationService.validatePasswordLength);
+            initInputField('pl-register-phone', validationService.validatePhone);
+            initInputField('pl-register-shipment-volume');
+            initInputField('pl-register-terms-and-conditions');
```

Comment:
Codacy found an issue: ['initInputField' was used before it was defined.](https://app.codacy.com/gh/packlink-dev/ecommerce_module_core/pullRequest?prid=5826400)